src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -122,6 +124,13 @@ public async Task InvokeAsync(HttpContext context)
 
                     ExceptionDispatchInfo.Capture(e).Throw();
                 }
+                finally
+                {
+                    hub.ConfigureScope(scope =>
```

Comment:
I wouldn't rely on the thing being in the scope.

We can hold the reference to the transaction in the body of this method, to make sure we're finishign the same instance we started here.

Reply:
I agree. If the middleware starts a span/transaction, it should have a direct reference to it and use it to call finish when the other HTTP handlers it wraps terminate.
I'd expect something along the lines of what Bruno wrote in [his comment](https://github.com/getsentry/sentry-dotnet/pull/633/files/c99635da62183de25b1e29c0e82f8802181a41cf#r533866281).